api/rbac/v1alpha1/annotations.go

```diff
@@ -6,28 +6,14 @@ const (
 	// Kargo.
 	AnnotationKeyManaged = "rbac.kargo.akuity.io/managed"
 
-	// AnnotationKeyOIDCEmails is an annotation key that can be set on a
-	// ServiceAccount to associate it with a list of email addresses. The
-	// annotation is used to grant permissions to the ServiceAccount based on
-	// the email claim from a user who authenticates using OIDC.
-	// The value of the annotation should be a comma-separated list.
-	AnnotationKeyOIDCEmails = "rbac.kargo.akuity.io/email"
-
-	// AnnotationKeyOIDCGroups is an annotation key that can be set on a
-	// ServiceAccount to associate it with a list of groups from an OIDC
-	// provider. The annotation is used to grant permissions to the
-	// ServiceAccount based on the groups claim from a user who authenticates
-	// using OIDC.
-	// The value of the annotation should be a comma-separated list.
-	AnnotationKeyOIDCGroups = "rbac.kargo.akuity.io/groups"
-
-	// AnnotationKeyOIDCSubjects is an annotation key that can be set on a
-	// ServiceAccount to associate it with a list of subjects from an OIDC
+	// AnnotationKeyOIDCPrefix is an annotation key prefix that can be set on a
+	// ServiceAccount to associate it with a list of claims from an OIDC
 	// provider. The annotation is used to grant permissions to the
-	// ServiceAccount based on the subject claim from a user who authenticates
+	// ServiceAccount based on the claim that is both provided in the suffix and from a user who authenticates
 	// using OIDC.
+	// e.g. rbac.kargo.akuity.io/claim.subs: '123,456'
 	// The value of the annotation should be a comma-separated list.
-	AnnotationKeyOIDCSubjects = "rbac.kargo.akuity.io/sub"
+	AnnotationKeyOIDCClaimNamePrefix = "rbac.kargo.akuity.io/claim."
```

Comment:
Nit: Just want to suggest some minor wordsmithing of the comment.

> // AnnotationKeyOIDCPrefixis the prefix of an annotation key that can be set on a 
// ServiceAccount to associate it with any user authenticated via OIDC and having
// the claim indicated by the full annotation key with any of the values indicated by
// the annotation. The value of the annotation may be either a scalar string value or a
// comma-separated list.

Reply:
Yes definitely makes more sense, fixed.